doc/internal/man7/OSSL_DEPRECATED.pod

```diff
@@ -0,0 +1,159 @@
+OPENSSL_NO_DEPRECATED_0_9_8, OSSL_DEPRECATEDIN_0_9_8, OSSL_DEPRECATEDIN_0_9_8_FOR,
+OSSL_DEPRECATED,
+OSSL_DEPRECATED_FOR
+- Macros to deprecating symbols and/or remove them
+
+=head1 SYNOPSIS
+
+ #include <openssl/macros.h>
+
+ OSSL_DEPRECATED(since)
+ OSSL_DEPRECATED_FOR(since, msg)
+
+ OPENSSL_NO_DEPRECATED_3_0
+ OSSL_DEPRECATEDIN_3_0
+ OSSL_DEPRECATEDIN_3_0_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_1_1_1
+ OSSL_DEPRECATEDIN_1_1_1
+ OSSL_DEPRECATEDIN_1_1_1_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_1_1_0
+ OSSL_DEPRECATEDIN_1_1_0
+ OSSL_DEPRECATEDIN_1_1_0_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_1_0_2
+ OSSL_DEPRECATEDIN_1_0_2
+ OSSL_DEPRECATEDIN_1_0_2_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_1_0_1
+ OSSL_DEPRECATEDIN_1_0_1
+ OSSL_DEPRECATEDIN_1_0_1_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_1_0_0
+ OSSL_DEPRECATEDIN_1_0_0
+ OSSL_DEPRECATEDIN_1_0_0_FOR(msg)
+
+ OPENSSL_NO_DEPRECATED_0_9_8
+ OSSL_DEPRECATEDIN_0_9_8
+ OSSL_DEPRECATEDIN_0_9_8_FOR(msg)
+
+=head1 DESCRIPTION
+
+Deprecation means to indicate that a symbol (usually a function, but can also
+be a define and less commonly a type or even variable) should not be used
+any more.
+OpenSSL supports multiple types of deprecation, including compile-time
+warnings, all the way to the removal of the implementation.
+This is controlled by two configurations parameters: C<--api=> to
+specify a version, and C<no-deprecated> to control actual removal.
+These are documented in the F<INSTALL.md> file provided with the source.
+As a third case, sometimes an item is deprecated, but it is still used
+within the OpenSSL software itself.
+
+The standard macros follow the same naming convention using the release
+version as a suffix, replacing the periods such as in B<1.1.0> with underscores,
+like B<1_1_0>. Because the 3.0 release introduces a new numbering scheme,
+in the text below we will indicate it as B<OSSL_DEPRECATEDIN_I<version>>
+
+Many modern compilers allow an annotation, or attribute, to be attached
+to a function declaration.
+If this is not supported by the compiler being used, they expand to
+nothing.
+This is done by using B<OSSL_DEPRECATEDIN_I<version>> or
+B<OSSL_DEPRECATEDIN_I<version>_FOR> as the first part of a function
+declaration.
+For example:
+
+ OSSL_DEPRECATEDIN_1_1_0 __owur int
+ HMAC_Init(HMAC_CTX *ctx, const void *key, int len, const EVP_MD *md);
+
+These declarations are in the public header files that can be found
+in the F<include/openssl> directory in the source distribution.
+
+With this declaration, any program that uses the deprecated function,
+assuming OpenSSL is configured to support it, will get a compile-time
+warning that HMAC_Init() is deprecated.
+
+To make it possible to hide the declaration completely, all
+deprecations should be wrapped inside an ifdef. For compatibility with
+other OpenSSL macros, a double negative must be used:
+
+ #ifndef OPENSSL_NO_DEPRECATED_1_1_0
+ OSSL_DEPRECATEDIN_1_1_0 __owur int
+ HMAC_Init(HMAC_CTX *ctx, const void *key, int len, const EVP_MD *md);
+ #endif
```

Comment:
I would like a visual cue, such as a visible section title, that leads the eye here when you want to remember how it should be done in a public header file.  (`deprecated.pod` has that)